### PR TITLE
Register IIF as cannonical function

### PR DESCRIPTION
**Pull request by kosinsky**
Register IIF as canonical function

### CODE REVIEW

File: src/Microsoft.OData.Core/UriParser/BuiltInUriFunctions.cs [hunk 1]
```diff
@@ -535,5 +536,41 @@ private static FunctionSignatureWithReturnType[] CreateMathFunctionSignatureArra
...
+            var primitiveTypeKinds = Enum.GetValues(typeof(EdmPrimitiveTypeKind)).Cast<EdmPrimitiveTypeKind>().Where(k => k != EdmPrimitiveTypeKind.None && k != EdmPrimitiveTypeKind.Stream);
+            var result = new List<FunctionSignatureWithReturnType>();
+            foreach (var kind in primitiveTypeKinds)
+            {
+                var argumentType = GetPrimitiveTypeReference(kind, false);
+                result.Add(new FunctionSignatureWithReturnType(
+                    argumentType,
+                    EdmCoreModel.Instance.GetBoolean(false),
+                    argumentType,
+                    argumentType));
```
Comment (StanislawSwierc) [inline src/Microsoft.OData.Core/UriParser/BuiltInUriFunctions.cs]: How this is going to work when two argument differ just by the nullability. For example, if the second argument is string and second is nullable string. Your function could use the nullable version as the output should this happen.

**Review by StanislawSwierc — CHANGES_REQUESTED** [review]
:clock1:

File: src/Microsoft.OData.Core/UriParser/BuiltInUriFunctions.cs (hunk 1 above)
Comment (kosinsky) [reply src/Microsoft.OData.Core/UriParser/BuiltInUriFunctions.cs]: If one if nullable and another is not then both will be considered nullable. And function will match to nullable, nullable signature

File: src/Microsoft.OData.Core/UriParser/BuiltInUriFunctions.cs [hunk 2]
```diff
@@ -535,5 +536,41 @@ private static FunctionSignatureWithReturnType[] CreateMathFunctionSignatureArra
...
+                argumentType = GetPrimitiveTypeReference(kind, true);
```
Comment (StanislawSwierc) [inline src/Microsoft.OData.Core/UriParser/BuiltInUriFunctions.cs]: >argumentType [](start = 16, length = 12)

can you call it argumentTypeNullable?

**Review by StanislawSwierc — APPROVED** [review]
:shipit:

**Pull request merged**